UOP1_Project/Assets/PlayerSpawnSystem/PlayerLevelChange.cs

```diff
@@ -0,0 +1,94 @@
+﻿using System.Linq;
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.SceneManagement;
+
+
+//The main "Spawn System" management class. Call the
+public class PlayerLevelChange : MonoBehaviour
+{
+    public static PlayerLevelChange main;
+    [SerializeField] List<PlayerSpawnPoint> spawnPoints = new List<PlayerSpawnPoint>(); //Contains all the spawn points in the current scene
+
+    void Awake()
+    {
+        //========== Singleton design stuff ==========
+        if (main == null)
+        {
+            DontDestroyOnLoad(gameObject);
+            main = this;
+        }
+        else if (main != this)
+        {
+            Destroy(gameObject);
+        }
+
+        //========== Initialization ==========
+        SceneManager.sceneLoaded += OnSceneLoaded;
+    }
+
+    void OnDestroy()
+    {
+        SceneManager.sceneLoaded -= OnSceneLoaded;
+    }
+
+    void OnSceneLoaded(Scene scene, LoadSceneMode mode)
+    {
+        ClearSpawnPoints();
+    }
+
+    void ClearSpawnPoints()
+    {
+        spawnPoints.Clear();
+    }
+
+    public void AddSpawnPoint(PlayerSpawnPoint sp)
+    {
+        spawnPoints.Add(sp);
+    }
+
+    //Teleports/'Spawns' the player at the specified spawn point in the current scene
+    public void SpawnPlayerAtSpawnPoint(Protagonist player, string spawnPointLabel)
+    {
+        PlayerSpawnPoint sp = spawnPoints.First(point => point.label == spawnPointLabel);
+        Spawn(player, sp.transform.position, sp.transform.rotation);
+    }
+
+    void Spawn(Protagonist player, Vector3 pos, Quaternion rot)
+    {
+        player.characterController.enabled = false; //Disable the controller due to it interfering with changing the position directly
+        player.transform.position = pos;
+        player.characterController.enabled = true;
```

Comment:
You should save the original state of the characterController and set it back to its orginal state rather than setting true directly. For all you know, this controller might have already been disabled and you just enabled it by accident.
```suggestion
        bool originalState = player.characterController.enabled;
        player.characterController.enabled = false; //Disable the controller due to it interfering with changing the position directly
        player.transform.position = pos;
        player.characterController.enabled = originalState;
```